types/foundry-vtt-api/types/apps/form.d.ts

```diff
@@ -0,0 +1,164 @@
+interface FormApplicationOptions extends ApplicationOptions {
+    /**
+     * Whether the application form is editable - if true, it's fields will
+     * be unlocked and the form can be submitted. If false, all form fields
+     * will be disabled and the form cannot be submitted. Default is true.
+     */
+    editable?: boolean;
+    /**
+     * Whether to automatically close the application when it's contained
+     * form is submitted. Default is true.
+     */
+    closeOnSubmit?: boolean;
+    /**
+     * Whether to automatically submit the contained HTML form when the
+     * application window is manually closed. Default is false.
+     */
+    submitOnClose?: boolean;
+    /**
+     * Whether to automatically submit the contained HTML form when an input
+     * or select element is changed. Default is false.
+     */
+    submitOnChange?: boolean;
+}
+
+/**
+ * An abstract pattern for defining an Application responsible for updating some object using an HTML form
+ *
+ * A few critical assumptions:
+ * 1) This application is used to only edit one object at a time
+ * 2) The template used contains one (and only one) HTML <form> as it's outer-most element
+ * 3) This abstract layer has no knowledge of what is being updated, so the implementation must define _updateObject
+ *
+ * @param object	Some object or entity which is the target to be updated.
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/foundry-vtt-api/types/apps/form.d.ts;
  Line: 33;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.